lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -6026,6 +6027,79 @@ void ModuleEmitter::emitHWModule(HWModuleOp module) {
   currentModuleOp = nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// Emitter for files & file lists.
+//===----------------------------------------------------------------------===//
+
+class FileEmitter : public EmitterBase {
+public:
+  explicit FileEmitter(VerilogEmitterState &state) : EmitterBase(state) {}
+
+  void emit(emit::FileOp op);
+  void emit(emit::FileListOp op);
+
+private:
+  void emitOp(emit::VerbatimOp op);
+};
+
+void FileEmitter::emit(emit::FileOp op) {
+  for (Operation &op : *op.getBodyBlock()) {
+    TypeSwitch<Operation *, void>(&op)
+        .Case<emit::VerbatimOp>([&](auto op) { emitOp(op); })
+        .Default([&](auto op) {
+          op->emitError("cannot emit operation to file");
+          state.encounteredError = true;
+        });
```

Comment:
The void return type on `TypeSwitch` may be implicit and not necessary to specify.